test/framework/ingress.go

```diff
@@ -23,238 +24,165 @@ func (s HostAndPort) Port() string {
-	err = f.kustomizeAndApply(dest, namespace, k8sContext)
-	require.NoError(t, err)
+	f.deployIngress(t, k8sContext, namespace, "solr-ingress.yaml", "solr", solrTemplate,
+		&ingressKustomization{solrServiceName, solrHostAndPort.Host()})
 }
 
 func (f *E2eFramework) DeployGraphIngresses(t *testing.T, k8sContext, namespace, graphServiceName string, graphHostAndPort HostAndPort) {
-	src := filepath.Join("..", "..", "test", "testdata", "ingress", "graph-ingress.yaml")
-	dest := filepath.Join("..", "..", "build", "test-config", "ingress", "graph", k8sContext)
+	f.deployIngress(t, k8sContext, namespace, "graph-ingress.yaml", "graph", graphTemplate,
+		&ingressKustomization{graphServiceName, graphHostAndPort.Host()})
+}
+
+func (f *E2eFramework) deployIngress(t *testing.T, k8sContext, namespace, sourceYaml, buildDir, template string, templateData interface{}) {
+	src := filepath.Join("..", "..", "test", "testdata", "ingress", sourceYaml)
+	dest := filepath.Join("..", "..", "build", "test-config", "ingress", buildDir, k8sContext)
 	_, err := utils.CopyFileToDir(src, dest)
 	require.NoError(t, err)
-	err = generateGraphIngressKustomization(k8sContext, namespace, graphServiceName, graphHostAndPort.Host())
+
+	err = generateKustomizationFile(fmt.Sprintf("ingress/%s/%s", buildDir, k8sContext), templateData, template)
 	require.NoError(t, err)
+
 	err = f.kustomizeAndApply(dest, namespace, k8sContext)
 	require.NoError(t, err)
 }
 
 func (f *E2eFramework) UndeployAllIngresses(t *testing.T, k8sContext, namespace string) {
 	options := kubectl.Options{Context: k8sContext, Namespace: namespace}
-	err := kubectl.DeleteAllOf(options, "IngressRoute")
-	assert.NoError(t, err)
-	err = kubectl.DeleteAllOf(options, "IngressRouteTCP")
+	err := kubectl.DeleteAllOf(options, "Ingress")
 	assert.NoError(t, err)
 }
 
-type ingressKustomization struct {
-	Namespace   string
-	ServiceName string
-	Host        string
-}
-
-func generateStargateIngressKustomization(k8sContext, namespace, serviceName, host string) error {
-	tmpl := `apiVersion: kustomize.config.k8s.io/v1beta1
+const stargateTemplate = `apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 resources:
 - stargate-ingress.yaml
-namespace: {{ .Namespace }}
 patches:
 - target:	
-    group: traefik.containo.us
-    version: v1alpha1
-    kind: IngressRoute
+    group: networking.k8s.io
+    version: v1
+    kind: Ingress
     name: cluster1-dc1-stargate-service-http-ingress
   patch: |-
     - op: replace
       path: /metadata/name
       value: "{{ .ServiceName }}-http-ingress"
+    - op: replace
+      path: /spec/rules/0/host
+      value: "{{ .Host }}"
+    - op: replace
+      path: /spec/rules/0/http/paths/0/backend/service/name
+      value: "{{ .ServiceName }}"
+    - op: replace
+      path: /spec/rules/0/http/paths/1/backend/service/name
+      value: "{{ .ServiceName }}"
+    - op: replace
+      path: /spec/rules/0/http/paths/2/backend/service/name
+      value: "{{ .ServiceName }}"
```

Comment:
I haven't looked in detail at what `patchesJson6902`, but with this version it's not needed, all the operations can be under `patch`.

Reply:
`patch` is newer and actually still applies the Json6902 style patch. For reference:

* https://github.com/kubernetes-sigs/kustomize/blob/master/examples/jsonpatch.md
https://stackoverflow.com/questions/63604579/what-is-the-difference-between-patches-vs-patchesjson6902-in-kustomize#63605421